Platform/ElkhartlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -315,6 +315,38 @@ FillGpioTable (
   return GpioTable;
 }
 
+
+/**
+  Configure the GPIO pins for GBEs: PCH TSN and Pse TSNs
+**/
+VOID
+ConfigureGbeGpio (
+  VOID
+  )
+{
+  SILICON_CFG_DATA            *SiCfgData;
+  MEMORY_CFG_DATA             *MemCfgData;
+
+  DEBUG ((DEBUG_INFO, "ConfigureGpio for Gbe\n"));
+  SiCfgData = (SILICON_CFG_DATA *)FindConfigDataByTag (CDATA_SILICON_TAG);
+  MemCfgData = (MEMORY_CFG_DATA *)FindConfigDataByTag (CDATA_MEMORY_TAG);
+
+  if (SiCfgData != NULL) {
+    if (SiCfgData->PchTsnEnable) {
+      ConfigureGpio (CDATA_NO_TAG, ARRAY_SIZE(mEhlPchTsnDeviceGpioTable), (UINT8*)mEhlPchTsnDeviceGpioTable);
+    }
+
+    if (MemCfgData != NULL && MemCfgData->PchPseEnable) {
+      if (SiCfgData->PchPseGbeEnable[0] > 0) {
+        ConfigureGpio (CDATA_NO_TAG, ARRAY_SIZE(mEhlPseTsn0DeviceGpioTable), (UINT8*)mEhlPseTsn0DeviceGpioTable);
+      }
+      if (SiCfgData->PchPseGbeEnable[1] > 0) {
+        ConfigureGpio (CDATA_NO_TAG, ARRAY_SIZE(mEhlPseTsn1DeviceGpioTable), (UINT8*)mEhlPseTsn1DeviceGpioTable);
```

Comment:
The changes look good to me. But often when corresponding FSP UPD is enabled, FSP should program these GPIO tables to make sure these GPIO pins works as expected. maybe check with FSP team if it is not done inside FSP.